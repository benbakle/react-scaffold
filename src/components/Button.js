import React from "react";
import { ThemeContextConsumer } from './../contexts/themes';

export default function Button(props) {
    return (
        <ThemeContextConsumer>
            {({ theme, toggleTheme }) => (
                <button onClick={toggleTheme}>
                    <span>Toggle to </span>
                    {
                        theme === "light" ?
                            <span role="img" aria-label="moon">ðŸŒš</span> :
                            <span role="img" aria-label="sun">ðŸŒž</span>
                    }
                </button>
            )}
        </ThemeContextConsumer>
    );
}