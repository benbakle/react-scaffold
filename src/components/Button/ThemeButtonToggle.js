import React, { useState } from "react";
import ThemeContext from '../../contexts/themes';

export default function ThemeButtonToggle(props) {
    const [count, setCount] = useState(0);

    const checkToggle = callback => {
        if (count >= 5)
            return window.location = 'https://www.youtube.com/watch?v=oHg5SJYRHA0';

        setCount(count + 1);
        callback();
    }

    return (
        <ThemeContext.Consumer>
            {({ theme, toggleTheme }) => (
                <button onClick={() => { checkToggle(toggleTheme) }}>
                    <span>Toggle to </span>
                    {
                        theme === "light" ?
                            <span role="img" aria-label="moon">ðŸŒš</span> :
                            <span role="img" aria-label="sun">ðŸŒž</span>
                    }
                </button>
            )}
        </ThemeContext.Consumer>
    );
}