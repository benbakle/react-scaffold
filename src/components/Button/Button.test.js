import React from 'react';
import { mount } from 'enzyme';
import ThemeContext from '../../contexts/themes';
import { Button } from './Button';

describe("The Button component", () => {
    let _component, _toggleThemeHasBeenCalled;

    beforeEach(() => {
        jest.resetModules();
        setComponentWithContext("light");
    });

    it('should match the light theme snapshot', () => {
        expect(_component.html()).toMatchSnapshot();
    });

    it('has a toggle button', () => {
        expect(_component.find("button").text()).toContain("Toggle to");
    });

    describe('when the context theme is light', () => {
        it("the toggle button shows the moon icon", () => {
            expect(_component.find('button [aria-label="moon"]').length).toBe(1);
        });
    });

    describe('when the toggle button is clicked', () => {
        it('calls to toggle theme', () => {
            _component.find("button").simulate("click");
            expect(_toggleThemeHasBeenCalled).toBeTruthy();
        });
    });

    describe('when the context theme is dark', () => {
        it('should match the dark theme snapshot', () => {
            expect(_component.html()).toMatchSnapshot();
        });

        it("the toggle button shows the sun icon", () => {
            setComponentWithContext("dark");
            expect(_component.find('button [aria-label="sun"]').length).toBe(1);
        });
    });

    function setComponentWithContext(theme) {
        _component = mount(
            <ThemeContext.Provider value={{ theme, toggleTheme: () => { _toggleThemeHasBeenCalled = true } }}>
                <Button />
            </ThemeContext.Provider>
        );
    }
});