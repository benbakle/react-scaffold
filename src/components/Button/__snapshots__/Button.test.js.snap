// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Button component should match the light theme snapshot 1`] = `"<button><span>Toggle to </span><span role=\\"img\\" aria-label=\\"moon\\">ðŸŒš</span></button>"`;

exports[`The Button component when the context theme is dark should match the dark theme snapshot 1`] = `"<button><span>Toggle to </span><span role=\\"img\\" aria-label=\\"moon\\">ðŸŒš</span></button>"`;
